tests/language-feature/types/is-on-type.slang

```diff
@@ -0,0 +1,45 @@
+
+//TEST(compute):COMPARE_COMPUTE_EX(filecheck-buffer=CHECK):-slang -compute
+
+// Test that `is` operator works on generic type param.
+
+//TEST_INPUT:ubuffer(data=[0 0 0 0], stride=4):out,name=outputBuffer
+RWStructuredBuffer<int> outputBuffer;
+
+interface I
+{
+
+}
+
+struct A : I
+{
+    uint i;
+};
+
+struct B : I
+{
+    float2 f2;
+};
+
+func test<T : I>(T t) -> int
```

Comment:
Just out of curiosity, what is this syntax starting with "func"?
This is not HLSL nor GLSL, is it?
I don't remember reading this syntax in the Slang user guide either.

Reply:
No, this is the "modern" syntax supported in Slang. We should add to the documentation that we support this.